component/upgrade/patch.sh

```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+set -e
+
+#Patch CRDS so that ArgoCD does not delete them during upgrade
+for crd in $CRDS_TO_PATCH; do
+	  exists=$(kubectl get crd "${crd}" --ignore-not-found)
```

Comment:
The indentation here is wrong. Looks like there's a tab and two spaces.

Also I'd go with something like this instead:
```suggestion
if ! kubectl get crd "${crd}" > /dev/null 2>&1; then
  >&2 echo "WARNING: Skipping 'blah': not found."
  continue
fi
```

This leverages the non-zero exitcode of the failed command instead of relying on an output.